File: app/src/main/java/com/anandm/scanner/di/AppComponent.java
package com.anandm.scanner.di;

import android.app.Application;


import com.anandm.scanner.ScannerApp;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.AndroidInjector;
import dagger.android.support.AndroidSupportInjectionModule;

@Singleton
@Component(
        modules = {
                AndroidSupportInjectionModule.class,
                ActivityBuilder.class
        }
)
public interface AppComponent extends AndroidInjector<ScannerApp> {

    void inject(ScannerApp app);

    @Component.Builder
    interface Builder {

        @BindsInstance
        Builder application(Application application);

        AppComponent build();
    }

}

File: app/src/main/java/com/anandm/scanner/di/AppComponent.kt
package com.anandm.scanner.di

import android.app.Application
import com.anandm.scanner.ScannerApp
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

@Singleton
@Component(
    modules = [
        AndroidSupportInjectionModule::class,
        ActivityBuilder::class
    ]
)
interface AppComponent : AndroidInjector<ScannerApp> {

    override fun inject(app: ScannerApp)

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: Application): Builder
        fun build(): AppComponent
    }

}